crt_portal/cts_forms/views.py

```diff
@@ -530,46 +531,38 @@ def get(self, request):
             'ids_count': ids_count,
             'show_warning': ids_count > 15,
             'all_ids_count': all_ids_count,
-            'assign_form': assign_form,
+            'bulk_actions_form': bulk_actions_form,
         }
         return render(request, 'forms/complaint_view/actions/index.html', output)
 
     def post(self, request):
-        assign_form = BulkAssign(request.POST)
+        bulk_actions_form = BulkActions(request.POST)
         return_url_args = request.POST.get('next', '')
         selected_all = request.POST.get('all', '') == 'all'
         confirm_all = request.POST.get('confirm_all', '') == 'confirm_all'
         ids = request.POST.get('ids', '').split(',')
 
-        if assign_form.is_valid():
-            assignee = assign_form.cleaned_data['assigned_to']
+        if bulk_actions_form.is_valid():
             if confirm_all:
                 requested_query = self.reconstruct_query(return_url_args)
             else:
                 requested_query = Report.objects.filter(pk__in=ids)
 
-            # update activity log _before_ we update the assignee so
-            # that we have access to the original assignee
-            for report in requested_query:
-                original = report.assigned_to or "None"
-                description = f'Updated from "{original}" to "{assignee}"'
-                add_activity(request.user, "Assigned to:", description, report)
-
-            number = requested_query.update(assigned_to=assignee)
-
-            description = f"{number} records have been assigned to {assignee}"
-            messages.add_message(request, messages.SUCCESS, description)
+            number = bulk_actions_form.update(requested_query, request.user)
+            description = bulk_actions_form.get_update_description()
+            message = f'{number} records have been updated: {description}'
```

Comment:
Small but handling the singular `record` case would be nice here 😄

Reply:
Sure thing, also updated the template (with the exception of "select all"). Does make me wonder if staff will use this function for one record instead of clicking on the full record.